inference/core/entities/responses/inference.py

```diff
@@ -272,3 +272,17 @@ class FaceDetectionPrediction(ObjectDetectionPrediction):
         alias="class", default="face", description="The predicted class label"
     )
     landmarks: Union[List[Point], List[Point3D]]
+
+
+def response_from_type(model_type, response_dict):
+    if model_type == "classification":
+        try:
+            return ClassificationInferenceResponse(**response_dict)
```

Comment:
Seems like this try/except logic could be hard to debug. I wonder if we could update get_model_type to differentiate `classification` from `multi-label-classification`, then keep track of it like a truly separate task (even though it's handled by the same model type). Not a show stopper here, just a thought.

Reply:
I agree -- plan is to later add a  `response_type` and corresponding factory to make this more bullet proof but gonna punt for now